themes/bootstrap3/templates/search/facet-list.phtml

```diff
@@ -28,36 +28,38 @@
+        <a href="<?=$urlBase . '&amp;facetpage=1&amp;facetsort=' . urlencode($key) . '&amp;contains=' . urlencode($contains) ?>" class="btn btn-default js-facet-sort<?php if($this->sort == $key): ?> active<?php endif; ?>" data-sort="<?=$key ?>"><?=$this->translate($sort) ?></a>
       <?php endforeach; ?>
     </div>
   </div>
 <?php endif; ?>
-<div class="lightbox-scroll full-facets">
+
+<form>
+  <?php if (method_exists($this->results->getParams(), 'setFacetContains')): ?>
+    <input class="ajax_param" data-name="contains" placeholder="<?=$this->transEscAttr('Filter')?>" class="form-control" type="text" value="<?=htmlspecialchars($this->contains)?>">
+  <?php endif; ?>
+  <?php $ajaxParams = ['facet' => $this->facet, 'facetexclude' => $this->exclude, 'facetop' => $this->operator, 'facetpage' => $this->page, 'facetsort' => $this->sort, 'urlBase' => $urlBase, 'searchAction' => $searchAction, 'ajaxLightbox' => $this->ajaxLightbox ? intval($this->ajaxLightbox) : intval($this->inLightbox)]; ?>
+  <input class="ajax_params" type="hidden" data-params="<?=$this->htmlSafeJsonEncode($ajaxParams)?>">
+</form>
+
+<?php $this->headScript()->appendFile('facets.js'); ?>
+
+<div id="facet-info-result" class="lightbox-scroll full-facets">
   <?php foreach ($this->sortOptions as $key => $sort): ?>
     <?php $active = $this->sort == $key; ?>
-    <div class="full-facet-list facet-group<?php if (!$active): ?> hidden<?php endif; ?>" id="facet-list-<?=$this->escapeHtmlAttr($key) ?>">
-      <?php if ($active): ?>
-        <?php if ($this->page > 1): ?>
-          <a href="<?=$urlBase . '&amp;facetpage=' . ($this->page - 1) . '&amp;facetsort=' . urlencode($this->sort) ?>" class="facet js-facet-prev-page" data-page="<?=($this->page + 1) ?>" data-sort="<?=$this->sort ?>" data-limit="<?=count($this->data) ?>" data-lightbox-ignore><?=$this->translate('prev_ellipsis') ?></a>
-        <?php endif; ?>
-        <?php foreach ($this->data as $item): ?>
-          <?=$this->render('Recommend/SideFacets/single-facet.phtml', [
-            'exclude' => $this->exclude,
-            'facet' => $item,
-            'group' => $this->facet,
-            'url' => $this->results->getUrlQuery(),
-            'urlBase' => $searchAction,
-          ]) ?>
-        <?php endforeach; ?>
-      <?php endif; ?>
-      <?php if ($this->anotherPage): ?>
-        <a href="<?=$urlBase . '&amp;facetpage=' . ($this->page + 1) . '&amp;facetsort=' . urlencode($key) ?>" class="facet js-facet-next-page" data-page="<?=($this->page + 1) ?>" data-sort="<?=$this->escapeHtmlAttr($key) ?>" data-lightbox-ignore>
-          <span class="text"><?=$this->translate('more_ellipsis') ?></span>
-        </a>
-      <?php endif; ?>
+    <div class="full-facet-list facet-group<?php if(!$active): ?> hidden<?php endif; ?>" id="facet-list-<?=$this->escapeHtmlAttr($key) ?>">
+      <!-- The content of this list will later be updated via AJAX, see facet-list-content.phtml -->
```

Comment:
Might make sense to reformat this as a single PHP block, to save on context-switching and to hide the comment from end user output:

```
<?php
  // The content of this list will later be updated via AJAX, see facet-list-content.phtml
  $params = $this->delegateParams;
  $params['urlBase'] = $urlBase;
  $params['searchAction'] = $searchAction;
  $params['inLightbox'] = $this->inLightbox;
?>
```